SQL join using interpolation with aliases

I am trying out the sql thing:
Scio v0.8.0-SNAPSHOT (after the alpha release)
The following code works:
```
    val s: SCollection[User] = thousand.users
    val a: SCollection[Artist] = thousand.artists

    val z = tsql"SELECT s.* FROM $s as s INNER JOIN $a as a ON $s.userId = $a.artistId"
      .as[User]
```

The following code doesn't
```
    val s: SCollection[User] = thousand.users
    val a: SCollection[Artist] = thousand.artists

    val z = sql"SELECT s.* FROM $s as s INNER JOIN $a as a ON $s.userId = $a.artistId"
      .as[User]
```

Looks like when using `sql` interpolator, aliasing "as s" doesn't work.

The following works:
```
    val s: SCollection[User] = thousand.users
    val a: SCollection[Artist] = thousand.artists

    val z = sql"SELECT SCOLLECTION_2.* FROM $s INNER JOIN $a ON $s.userId = $a.artistId"
      .as[User]
```

Bioler plate [here](https://gist.github.com/anish749/fd12ee02f32c7df7e2b08ccc0b89ed55#file-qscio-scala-L63-L72)

case class definitions [here](https://gist.github.com/anish749/fd12ee02f32c7df7e2b08ccc0b89ed55#file-qscio-scala-L158-L173)

```
Exception in thread "main" java.lang.RuntimeException: 
SqlValidatorException: Table 'SCOLLECTION_2' not found

Query:
SELECT s.* FROM  SCOLLECTION_2 as s INNER JOIN  SCOLLECTION_3 as a ON  SCOLLECTION_2.userId =  SCOLLECTION_3.artistId 


schema of SCOLLECTION_2:
┌──────────────────────────────────────────┬──────────┬──────────┐
│ NAME                                     │ TYPE     │ NULLABLE │
├──────────────────────────────────────────┼──────────┼──────────┤
│ userId                                   │ STRING   │ NO       │
│ userIdLong                               │ INT64    │ NO       │
│ fields                                   │ STRING[] │ NO       │
└──────────────────────────────────────────┴──────────┴──────────┘
        

schema of SCOLLECTION_3:
┌──────────────────────────────────────────┬──────────┬──────────┐
│ NAME                                     │ TYPE     │ NULLABLE │
├──────────────────────────────────────────┼──────────┼──────────┤
│ artistId                                 │ STRING   │ NO       │
│ artistIdLong                             │ INT64    │ NO       │
│ artistName                               │ STRING   │ NO       │
│ fields                                   │ STRING[] │ NO       │
└──────────────────────────────────────────┴──────────┴──────────┘
        
Query result schema (inferred) is unknown.
Expected schema:
┌──────────────────────────────────────────┬──────────┬──────────┐
│ NAME                                     │ TYPE     │ NULLABLE │
├──────────────────────────────────────────┼──────────┼──────────┤
│ userId                                   │ STRING   │ NO       │
│ userIdLong                               │ INT64    │ NO       │
│ fields                                   │ STRING[] │ NO       │
└──────────────────────────────────────────┴──────────┴──────────┘
        
	at com.spotify.scio.sql.SqlSCollection2$$anonfun$queryAs$2.apply(Query2.scala:134)
	at com.spotify.scio.sql.SqlSCollection2$$anonfun$queryAs$2.apply(Query2.scala:134)
	at scala.util.Either.fold(Either.scala:189)
	at com.spotify.scio.sql.SqlSCollection2.queryAs(Query2.scala:134)
	at com.spotify.scio.sql.SQLBuilders$$anon$2.as(SQLBuilders.scala:60)
	at benchmark.ExpressionJoin$.$anonfun$main$1(ExpressionJoin.scala:12)
	at benchmark.ExpressionJoin$.$anonfun$main$1$adapted(ExpressionJoin.scala:7)
	at qscio.DataflowBenchmark.benchLocal(DataflowBenchmark.scala:68)
	at qscio.DataflowBenchmark.benchLocal$(DataflowBenchmark.scala:63)
	at benchmark.ExpressionJoin$.benchLocal(ExpressionJoin.scala:6)
	at benchmark.ExpressionJoin$.main(ExpressionJoin.scala:7)
	at benchmark.ExpressionJoin.main(ExpressionJoin.scala)
Caused by: org.apache.beam.sdk.extensions.sql.impl.ParseException: Unable to parse query SELECT s.* FROM  SCOLLECTION_2 as s INNER JOIN  SCOLLECTION_3 as a ON  SCOLLECTION_2.userId =  SCOLLECTION_3.artistId 
	at org.apache.beam.sdk.extensions.sql.impl.BeamSqlEnv.parseQuery(BeamSqlEnv.java:145)
	at org.apache.beam.sdk.extensions.sql.SqlTransform.expand(SqlTransform.java:111)
	at org.apache.beam.sdk.extensions.sql.SqlTransform.expand(SqlTransform.java:79)
	at org.apache.beam.sdk.Pipeline.applyInternal(Pipeline.java:537)
	at org.apache.beam.sdk.Pipeline.applyTransform(Pipeline.java:488)
	at org.apache.beam.sdk.values.PCollectionTuple.apply(PCollectionTuple.java:261)
	at com.spotify.scio.sql.SqlSCollection2.query(Query2.scala:116)
	at com.spotify.scio.sql.SqlSCollection2.query(Query2.scala:105)
	at com.spotify.scio.sql.SqlSCollection2.queryAs(Query2.scala:131)
	... 8 more
Caused by: org.apache.beam.repackaged.beam_sdks_java_extensions_sql.org.apache.calcite.tools.ValidationException: org.apache.beam.repackaged.beam_sdks_java_extensions_sql.org.apache.calcite.runtime.CalciteContextException: From line 1, column 72 to line 1, column 84: Table 'SCOLLECTION_2' not found
	at org.apache.beam.repackaged.beam_sdks_java_extensions_sql.org.apache.calcite.prepare.PlannerImpl.validate(PlannerImpl.java:190)
	at org.apache.beam.sdk.extensions.sql.impl.BeamQueryPlanner.convertToBeamRel(BeamQueryPlanner.java:123)
	at org.apache.beam.sdk.extensions.sql.impl.BeamSqlEnv.parseQuery(BeamSqlEnv.java:143)
	... 16 more
```

--- a/scio-core/src/main/scala/com/spotify/scio/sql/Sql.scala
+++ b/scio-core/src/main/scala/com/spotify/scio/sql/Sql.scala
@@ -34,6 +34,7 @@ import org.apache.beam.sdk.extensions.sql.meta.provider.{ReadOnlyTableProvider, 
 
 import scala.util.Try
 import scala.collection.JavaConverters._
+import org.apache.commons.lang3.exception.ExceptionUtils
 
 object Sql extends SqlSCollections {
 
@@ -65,6 +66,8 @@ object Sql extends SqlSCollections {
 }
 
 private object Queries {
+  private[this] val TableNotFound = raw"(SqlValidatorException: Table '(.*)' not found)".r
+  private[this] val UnknownIdentifier = raw"(SqlValidatorException: Unknown identifier '(.*)')".r
 
   private[this] def parseQuery(
     query: String,
@@ -125,7 +128,14 @@ private object Queries {
       .toEither(schema(query, inferredSchemas, udfs))
       .left
       .map { ex =>
-        val mess = org.apache.commons.lang3.exception.ExceptionUtils.getRootCauseMessage(ex)
+        val mess = ExceptionUtils.getRootCauseMessage(ex) match {
+          case TableNotFound(msg, _) =>
+            s"$msg\nHint: incorrect table aliases are being used!"
+          case UnknownIdentifier(msg, _) =>
+            s"$msg\nHint: incorrect table aliases are being used!"
+          case msg =>
+            msg
+        }
 
         s"""
            |$mess

